docs/validator.go

```diff
@@ -0,0 +1,167 @@
+ * (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/a8m/mark"
+	"github.com/gernest/wow"
+	"github.com/gernest/wow/spin"
+	"github.com/minio/cli"
+)
+
+func init() {
+	// Validate go binary.
+	if _, err := exec.LookPath("go"); err != nil {
+		panic(err)
+	}
+}
+
+var globalFlags = []cli.Flag{
+	cli.StringFlag{
+		Name:  "m",
+		Value: "API.md",
+		Usage: "Path to markdown api documentation.",
+	},
+	cli.StringFlag{
+		Name:  "t",
+		Value: "checker.go.tpl",
+		Usage: "Template used for generating the programs.",
+	},
+	cli.IntFlag{
+		Name:  "skip",
+		Value: 2,
+		Usage: "Skip entries before validating the code.",
+	},
+}
+
+func validatorAction(ctx *cli.Context) error {
+	if !ctx.IsSet("m") || !ctx.IsSet("t") {
+		return nil
+	}
+	docPath := ctx.String("m")
+	var err error
+	docPath, err = filepath.Abs(docPath)
+	if err != nil {
+		return err
+	}
+	data, err := ioutil.ReadFile(docPath)
+	if err != nil {
+		return err
+	}
+
+	templatePath := ctx.String("t")
+	templatePath, err = filepath.Abs(templatePath)
+	if err != nil {
+		return err
+	}
+
+	skipEntries := ctx.Int("skip")
+	m := mark.New(string(data), &mark.Options{
+		Gfm: true, // Github markdown support is enabled by default.
+	})
+
+	t, err := template.ParseFiles(templatePath)
+	if err != nil {
+		return err
+	}
+
+	tmpDir, err := ioutil.TempDir("", "md-verifier")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpDir)
+
+	entryN := 1
+	for i := mark.NodeText; i < mark.NodeCheckbox; i++ {
+		if mark.NodeCode != mark.NodeType(i) {
+			m.AddRenderFn(mark.NodeType(i), func(node mark.Node) (s string) {
+				return ""
+			})
+			continue
+		}
+		m.AddRenderFn(mark.NodeCode, func(node mark.Node) (s string) {
+			p, ok := node.(*mark.CodeNode)
+			if !ok {
+				return
+			}
+			p.Text = strings.NewReplacer("&lt;", "<", "&gt;", ">", "&quot;", `"`, "&amp;", "&").Replace(p.Text)
+			if skipEntries > 0 {
+				skipEntries--
+				return
+			}
+
+			testFilePath := filepath.Join(tmpDir, fmt.Sprintf("%d.go", entryN))
+			testBinPath := filepath.Join(tmpDir, fmt.Sprintf("%d", entryN))
+			w, werr := os.Create(testFilePath)
+			if werr != nil {
+				panic(werr)
+			}
+			t.Execute(w, p)
+			w.Sync()
+			w.Close()
+			entryN++
+
+			cmdArgs := []string{"-s", "-w", "-l", testFilePath}
+			exec.Command("gofmt", cmdArgs...).Run()
```

Comment:
You could have a function or closure hide this logic here.  This way you could have cmdArgs globally.  If available I prefer long option than short option for readability (if available)
```go
function runGofmt(string testFilePath) {
    ...
}
```